doc/api/package_json.md

```diff
@@ -0,0 +1,649 @@
+# Modules: Package configuration
+
+<!-- type=misc -->
+> Stability: 1 - Experimental
```

Comment:
This seems a little weird to me given the title of the document "The package.json file".  I was reading this as the package.json is at the experimental phase

Reply:
Understandable, but that's just how it looks in the source code. Compare to the ESM page.

![image](https://user-images.githubusercontent.com/17770407/90566726-f9b4fd00-e176-11ea-87e2-475e2002fc23.png)

Reply:
Then maybe a change to the title to "Modules: ECMAScript Modules Package Configuration"?   not really tied to that, though

Reply:
I was envisioning that we would rename the file `doc/api/modules_packages.md` and title it `Modules: Packages`.

`esm.md` could be likewise renamed `modules_esm.md` and titled `Modules: ECMAScript modules`.

`modules.md` could become `modules_commonjs.md` and titled `Modules: CommonJS modules`.

Work would need to be done to create redirect links for all the old headings.

Regarding stability, only the ECMAScript modules page would have a page-level experimental warning. On the packages page, individual sections should have their own stability levels listed (as not everything related to packages is experimental, such as `"main"` etc.).

Reply:
Frankly I feel like this page describes a concept perpendicular to what the community understand by "packages", and would be clearer if named "Export Configuration", "Exported Modules", or something similar (ie anything that isn't "package").

> Elsewhere we had discussed making this page about package authoring in general, not specific to just `package.json`. That way the page is free to include sections that aren't necessarily `package.json` fields.

@GeoffreyBooth I understand, but then the title and perspective can be modified when it comes to that, if ever. In its current state, I really think it will be confusing for people landing on this page and seeing no reference to dependencies... and given that exports are a power user feature, I find likely that more people will land on this page thinking about regular package management.

Reply:
> `esm.md` could be likewise renamed `modules_esm.md` and titled `Modules: ECMAScript modules`.
> 
> `modules.md` could become `modules_commonjs.md` and titled `Modules: CommonJS modules`.

That's covered by #34663. We can't rename the pages without breaking existing links so that's been ruled out.

> Work would need to be done to create redirect links for all the old headings.

That'd be ideal, unfortunately I don't have the bandwidth to do that. If someone wants to pick it up, you're very welcome.

> I was envisioning that we would rename the file `doc/api/modules_packages.md` and title it `Modules: Packages`.

The first section is called `Packages` already, that seems a bit redundant, what are your thoughts?

Reply:
> Frankly I feel like this page describes a concept perpendicular to what the community understand by "packages"

This page aims to cover all the topics that are related to the module system, but are not ESM nor CJS specific. It's doesn't have to be "packages" indeed, I'm definitely open to suggestions.

On a side note, npm defines a package as `a file or directory that is described by a package.json file` ([source](https://docs.npmjs.com/about-packages-and-modules)). That's unfortunate npm and Node.js uses the same terms with very different definitions, and that's a good point for not using it as the title of the document…

Reply:
A package is a folder with a package.json in it. I'm not sure how this is ambigous between any of the definitions?

Reply:
A package may be a file for npm. But OK let's not get distracted by that.

I've tried to address the comments in my last commit, PTAL.